contrib/bin/strip_dup_libs.pl

```diff
@@ -52,58 +52,10 @@
         }
 }
 
-#
-# Move system library paths to the end of the link line.  This way,
-# there should be less risk of libmesh linking against something in
-# /usr/lib by accident.
-#
-my @cleaned_up_libs_fourth;
-my @system_lib_paths;
-foreach( @cleaned_up_libs_third ) {
-    $_ = remove_rel_paths($_);
-    # Discard libary paths starting with /usr.  This includes e.g.
-    # /usr/lib and /usr/local/lib, which are assumed to be system
-    # library paths.
-    if( ($_=~/-L\/usr/) )
-    {
-        push @system_lib_paths, $_;
-    }
-    # Discard -Wl,* flags starting with /usr
-    elsif( ($_=~/-Wl,[^\/]*\/usr/) )
-    {
-        push @system_lib_paths, $_;
-    }
-    # Discard library paths starting with /lib. This directory is not
-    # even present on OSX and it is typically fairly empty on Linux,
-    # but it may be present on other UNIX systems.
-    elsif( ($_=~/-L\/lib/) )
-    {
-        push @system_lib_paths, $_;
-    }
-    # Discard -Wl,* flags starting with /lib
-    elsif( ($_=~/-Wl,[^\/]*\/lib/) )
-    {
-        push @system_lib_paths, $_;
-    }
-    # Otherwise, keep the path.
-    else
-    {
-        push @cleaned_up_libs_fourth, $_;
-    }
-}
-
-#
-# Push system lib paths onto the end of the list
```

Comment:
We definitely can't just remove this bit about pushing system library paths to the end of the link line. It was added to fix issues on systems with multiple MPIs installed. If `-L/usr/lib` comes too early on the link line, then we run the risk of compiling against MPI "A" and then linking against MPI "B", which generally causes a runtime error/crash. 

The part about "discarding" library paths that start with `/usr/lib` and `/lib` maybe isn't necessary... I'm having trouble remembering what triggered the need for that off the top of my head.

Reply:
That's a good point. Would modifying this to discard `/usr/lib/`, `/usr/`, and `/lib/` (rather than using them for regex matching) work?

Reply:
Now that I read the code again I think I was thrown off by the "Discard" in the comments. The `-L/usr/lib` et al are not being dropped by this code, they are just being moved to the end of the link line.

In other words, I don't think this script should have *removed* `-L/usr/lib/openmpi/` from your link line ... it should have just moved it to the end. If it did remove it entirely then that seems like a bug we could maybe fix (I also am not great at perl).

Reply:
You are right - this script moves `-L/usr/lib/openmpi/` to the end of the link line, which then causes problems because the link line is now `-lmpi_mpifh [...] -L/usr/lib/openmpi/` which isn't going to work.

Reply:
Ahhh, now I think I'm starting to get it. This is a bit tricker of a solution, but it seems to me that we need to re-order things so that we end up with 
```
[all -L flags, with -L/usr/lib etc. at end] ... [all -l flags]
```
instead.

Reply:
I think that would work.

Reply:
I don't know perl but I'll try to implement that in the script.